homeassistant/components/deluge/sensor.py

```diff
@@ -67,92 +66,70 @@
 
     def __init__(
-        self, deluge_client, client_name, description: SensorEntityDescription
-    ):
+        self,
+        coordinator: DelugeDataUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
         """Initialize the sensor."""
+        super().__init__(coordinator)
         self.entity_description = description
-        self.client = deluge_client
-        self.data = None
-
-        self._attr_available = False
-        self._attr_name = f"{client_name} {description.name}"
-
-    def update(self):
-        """Get the latest data from Deluge and updates the state."""
-
-        try:
-            self.data = self.client.call(
-                "core.get_session_status",
-                [
-                    "upload_rate",
-                    "download_rate",
-                    "dht_upload_rate",
-                    "dht_download_rate",
-                ],
-            )
-            self._attr_available = True
-        except FailedToReconnectException:
-            _LOGGER.error("Connection to Deluge Daemon Lost")
-            self._attr_available = False
-            return
-
-        upload = self.data[b"upload_rate"] - self.data[b"dht_upload_rate"]
-        download = self.data[b"download_rate"] - self.data[b"dht_download_rate"]
-
-        sensor_type = self.entity_description.key
-        if sensor_type == "current_status":
-            if self.data:
-                if upload > 0 and download > 0:
-                    self._attr_native_value = "Up/Down"
-                elif upload > 0 and download == 0:
-                    self._attr_native_value = "Seeding"
-                elif upload == 0 and download > 0:
-                    self._attr_native_value = "Downloading"
-                else:
-                    self._attr_native_value = STATE_IDLE
-            else:
-                self._attr_native_value = None
-
-        if self.data:
-            if sensor_type == "download_speed":
-                kb_spd = float(download)
-                kb_spd = kb_spd / 1024
-                self._attr_native_value = round(kb_spd, 2 if kb_spd < 0.1 else 1)
-            elif sensor_type == "upload_speed":
-                kb_spd = float(upload)
-                kb_spd = kb_spd / 1024
-                self._attr_native_value = round(kb_spd, 2 if kb_spd < 0.1 else 1)
+        self._attr_name = f"{coordinator.config_entry.title} {description.name}"
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{description.key}"
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        if self.coordinator.data:
+            data = self.coordinator.data[Platform.SENSOR]
+            upload = data[b"upload_rate"] - data[b"dht_upload_rate"]
+            download = data[b"download_rate"] - data[b"dht_download_rate"]
+            if self.entity_description.key == "current_status":
+                if data:
```

Comment:
Add a check above at the top of the body if `data` is None, where we return None, instead.